services/dask-sidecar/src/simcore_service_dask_sidecar/utils.py

```diff
@@ -14,6 +14,23 @@ def wrap_async_call(fct: Awaitable) -> ...:
     return asyncio.get_event_loop().run_until_complete(fct)
 
 
+def cluster_id() -> Optional[str]:
+    """Returns the cluster id this docker engine belongs to, if any"""
+
+    async def async_get_engine_cluster_id() -> Optional[str]:
+        async with aiodocker.Docker() as docker:
+            docker_system_info = await docker.system.info()
+        node_labels = docker_system_info.get("Labels", [])
+        node_labels_dict = {}
+        for entry in node_labels:
+            key, value = str(entry).split("=", maxsplit=2)
+            if key and value:
```

Comment:
since you only want the ``cluster_id``, why fill the entire dict? 
```python
for entry in node_labels:
    try:
        key, value = str(entry).split("=", maxplit=2)
        if key=="cluster_id" and value:
             return value
    except KeyValue: # e.g. when not enough values to unpack
       log.warning("Mmmhhh something is fishy with docker labels: does %s sounds ok to you?", entry) 
return None
```